workloads/devstack-xenial.yaml

```diff
@@ -0,0 +1,55 @@
+---
+inherits: xenial
+needs_nested_vm: true
+vm:
+  mem_mib: 6144
+  cpus: 4
+  ports:
+  - host: 55580
+    guest: 80
+...
+---
+{{- define "ENV" -}}
+{{proxyVars .}}
+{{- print " DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true " -}}
+{{end}}
+{{ define "DEVSTACKPATH" }}{{ if $.MountPath "devstack" }}{{$.MountPath "devstack"}}{{else}}/home/{{.User}}/devstack{{end}}{{end}}
+#cloud-config
+write_files:
+ - content: |
+     [url "http://"]
+       insteadOf=git://
+   path: /home/{{.User}}/.gitconfig
+   permissions: '0755'
+
+runcmd:
+  - chown -R {{.User}}:{{.User}} /home/{{.User}}/
+
+{{- if not (.MountPath "devstack")}}
+  - {{beginTask . "Download devstack" }}
+  - sudo -u {{.User}} {{template "ENV" .}} git clone http://git.openstack.org/openstack-dev/devstack {{template "DEVSTACKPATH" .}}
+  - {{endTaskCheck .}}
+
+# Once dyamic IP provisioning is available, the HOST_IP can be populated with
+# it against the current static one.
+  - {{beginTask . "Create conf file" }}
+  - (printf "%s\n"
+    "[[local|localrc]]"
+    "HOST_IP=127.0.0.1"
+    "ADMIN_PASSWORD=secret"
+    "DATABASE_PASSWORD=\$ADMIN_PASSWORD"
+    "RABBIT_PASSWORD=\$ADMIN_PASSWORD"
+    "SERVICE_PASSWORD=\$ADMIN_PASSWORD"
+    ) > {{template "DEVSTACKPATH" .}}/local.conf
+  - {{endTaskCheck .}}
+
+  - chown -R {{.User}};{{.User}} {{template "DEVSTACKPATH" .}}/local.conf
+{{ end }}
+
+  - {{beginTask . "Stacking the system" }}
+  - sudo su -c "cd {{template "DEVSTACKPATH" .}} && nohup ./stack.sh > /home/{{.User}}/stack.log &" {{.User}}
```

Comment:
Should this be sudo -u {{.User}} or does this command really need to be run as root?  If that's the case you don't actually need sudo  su -c here at all.  All the commands in run_cmd run as root.

Reply:
Dumped the sudo and the command is run as {{.User}} now. Thanks for the review.